docs/core/deploying/deploy-with-cli.md

```diff
@@ -1,201 +1,154 @@
+### Native dependencies
 
-      ```xml
-      <ItemGroup>
-        <PackageReference Include="Newtonsoft.Json" Version="10.0.2" />
-      </ItemGroup>
-      ```
+If your app has some sort of native dependency, it may not run on a different operating system. For example, if you (on Windows) referenced a library that used the native Win32 API, your app wouldn't run on macOS or Linux. You would need to provide platform-specific code, and compile an executable for each platform. 
 
-1. If you haven't already, download the NuGet package containing the third-party dependency. To download the package, execute the `dotnet restore` ([see note](#dotnet-restore-note)) command after adding the dependency. Because the dependency is resolved out of the local NuGet cache at publish time, it must be available on your system.
+Consider also that if a library you referenced has a native dependency, your app also may not run on every platform. However, it's possible that a NuGet package you're referencing has included platform-specific versions, handling the required native dependencies for you.
 
-Note that a framework-dependent deployment with third-party dependencies is only as portable as its third-party dependencies. For example, if a third-party library only supports macOS, the app isn't portable to Windows systems. This happens if the third-party dependency itself depends on native code. A good example of this is [Kestrel server](/aspnet/core/fundamentals/servers/kestrel), which requires a native dependency on [libuv](https://github.com/libuv/libuv). When an FDD is created for an application with this kind of third-party dependency, the published output contains a folder for each [Runtime Identifier (RID)](../rid-catalog.md) that the native dependency supports (and that exists in its NuGet package).
+When distributing an app with native dependencies, you may need to use the `dotnet publish -r <RID>` switch to specify the target platform you want to publish for. For a list of runtime identifiers, see [Runtime Identifier (RID) catalog](..\rid-catalog.md).
 
-## <a name="simpleSelf"></a> Self-contained deployment without third-party dependencies
+More information about platform-specific binaries is covered in the [Framework-dependent executable](#framework-dependent-executable) and [Self-contained deployment](#self-contained-deployment) sections.
 
-Deploying a self-contained deployment without third-party dependencies involves creating the project, modifying the *csproj* file, building, testing, and publishing the app. A simple example written in C# illustrates the process. The example shows how to create a self-contained deployment using the [dotnet utility](../tools/dotnet.md) from the command line.
+## Sample app
 
-1. Create a directory for the project.
+If you would like to quickly test publishing an app, but don't have an app available, here is a small app to play with. The app is created by running the following commands in your terminal:
```

Comment:
This can be much shorter:

> You can use the following app to exploring CLI publishing commands, or any application you've created.

Reply:
Nit in BIll's suggestion

exploring -> explore

I don't think "or any application you've created" is adding much